diff --git a/dist/index.mjs b/dist/index.mjs
index 370b0143757e0e4352e0d447e56641dbddd1f45d..c1c8515c255575618170e014eb0c1b9074d8c0b7 100644
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -63716,7 +63716,11 @@ function filterEntryWorkerConfig(config$2) {
 }
 function customizeWorkerConfig(options) {
 	const configResult = typeof options.configCustomizer === "function" ? "entryWorkerConfig" in options ? options.configCustomizer(options.workerConfig, { entryWorkerConfig: filterEntryWorkerConfig(options.entryWorkerConfig) }) : options.configCustomizer(options.workerConfig) : options.configCustomizer;
-	if (configResult) return defu(configResult, options.workerConfig);
+	if (configResult) {
+		const __merged = defu(configResult, options.workerConfig);
+		if (Array.isArray(__merged.compatibility_flags)) __merged.compatibility_flags = [...new Set(__merged.compatibility_flags)];
+		return __merged;
+	}
 	return options.workerConfig;
 }
 /**
